fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2790,8 +2785,9 @@ private Bundle addLinks(FHIRPagingContext context, Bundle responseBundle, String
 
         // If for search with _summary=count or pageSize == 0, then don't add previous and next links.
         if (!SummaryValueSet.COUNT.equals(summaryParameter) && context.getPageSize() > 0) {
-            int nextPageNumber = context.getPageNumber() + 1;
-            if (nextPageNumber <= context.getLastPageNumber()) {
+            int nextPageNumber = Math.max(context.getPageNumber() + 1, 1);
```

Comment:
Can getPageNumber() ever be negative?

Reply:
Yes, if the user passes in _page=-500, that's what getPageNumber will return here.  The idea is for nextPage to point to page=1, since that is the first valid page.  Rather than nextPage=-499, which will take the caller a while to page back to the valid page numbers.